ci-wrappers/pythonpreview/wrapper/python_glue/internal_device_glue_async.py

```diff
@@ -68,7 +70,36 @@ def wait_for_c2d_message(self):
         return message_to_object(message)
 
     def roundtrip_method_call(self, methodName, requestAndResponse):
-        raise NotImplementedError()
+       # receive method request
+        print("Waiting for method request")
+        request = async_helper.run_coroutine_sync(self.client.receive_method_request(methodName))
+        print("Method request received")
+
+        # verify name and payload
+        expected_name = methodName
+        expected_payload = requestAndResponse.request_payload['payload']
+        if (request.name == expected_name):
+            if (request.payload == expected_payload):
+                print("Method name and payload matched. Returning response")
+                resp_status = requestAndResponse.status_code
+                resp_payload = requestAndResponse.response_payload
+            else:
+                print("Request payload doesn't match")
+                print("expected: " + expected_payload)
+                print("received: " + request.payload)
+                resp_status = 500
+                resp_payload = None
+        else:
+            print("Method name doesn't match")
+            print("expected: '" + expected_name + "'")
+            print("received: '" + request.name + "'")
+            resp_status = 404
+            resp_payload = None
```

Comment:
These 20 lines get reused 4 times.  I smell a helper-function.

Reply:
Is this that different from any other repetition of code between the four glue files though? Do we want to make helpers for all of them?

---
In reply to: [287466566](https://github.com/Azure/iot-sdks-e2e-fx/pull/71#discussion_r287466566) [](ancestors = 287466566)